### PR TITLE
Fix logical structure for scanned maps

### LINKED ISSUE #2377: Logical structure does not work for MapSets
Because of the way in which ScannedMaps are modeled, the structure manager saves the logical structure for MapSets in way that generates broken manifests. Specifically, the canvas id for a range with a ScannedMap does not match the id of the corresponding canvas.

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #2377

### CODE REVIEW

File: app/resources/scanned_maps/scanned_maps_controller.rb [hunk 1]
```diff
@@ -3,4 +3,13 @@ class ScannedMapsController < ScannedResourcesController
...
   include GeoblacklightDocumentController
   self.resource_class = ScannedMap
+
+  # View the structural metadata for a given repository resource
+  def structure
+    @change_set = change_set_class.new(find_resource(params[:id])).prepopulate!
+    @logical_order = (Array(@change_set.logical_structure).first || Structure.new).decorate
+    members = Wayfinder.for(@change_set.resource).logical_structure_members
+    @logical_order = WithProxyForObject.new(@logical_order, members)
+    authorize! :structure, @change_set.resource
```
Comment (escowles) [inline app/resources/scanned_maps/scanned_maps_controller.rb]: Should the `authorize!` call come as early as possible (presumably right after the @change_set is instantiated)?

File: app/resources/scanned_maps/scanned_map_wayfinder.rb [hunk 2]
```diff
@@ -34,4 +34,21 @@ def members_with_parents
...
+
+  # To display and save logical order correctly, this method replaces ScannedMap members that have no ScannedMap children
+  # with their corresponding geo member (TIFF) FileSet. Geo resources are unusual in that each resource has one corresponding
+  # geo member. An atlas with multiple maps, for example, will be a set of multiple ScannedMaps with each resource corresponding
+  # to a single map. ScannedMap members with ScannedMap children are analogous to a volume in a multi-volume work.
+  def logical_structure_members
+    @logical_structure_members ||= begin
+      members_with_parents.map do |member|
+        decorator = member.decorate
+        if decorator.respond_to?(:scanned_map_members) && decorator.scanned_map_members.empty?
```
Comment (escowles) [inline app/resources/scanned_maps/scanned_map_wayfinder.rb]: Pulling this out to a separate method would probably let you avoid the clunky `||= begin` syntax.

File: app/resources/scanned_maps/scanned_maps_controller.rb (hunk 1 above)
Comment (eliotjordan) [reply app/resources/scanned_maps/scanned_maps_controller.rb]: Makes sense. This method was copied from the scanned resource controller and modified slightly. I can update that one as well.

File: app/resources/scanned_maps/scanned_map_wayfinder.rb (hunk 2 above)
Comment (eliotjordan) [reply app/resources/scanned_maps/scanned_map_wayfinder.rb]: Yeah, it's it a little gnarly. I'll refactor.

Comment (eliotjordan) [reply app/resources/scanned_maps/scanned_map_wayfinder.rb]: Refactored to a new method, although the comment doesn't show as `outdated` for some reason.

**Pull request merged**